src/nflxmultisubs.js

```diff
@@ -14,14 +14,45 @@ const hookJsonParseAndAddCallback = function(_window) {
       if (result && result.result && result.result.movieId) {
           const movieId = result.result.movieId
           //console.log(`Intercepted manifest ${movieId}`);
-          window.__NflxMultiSubs.updateManifest(result.result);
+          // clone manifest so we can modify it w/o breaking player
+          let clonedManifest = _parse.call(JSON, ...args).result;
+          window.__NflxMultiSubs.updateManifest(clonedManifest);
+          if (gRenderOptions.secondaryLanguageMode === 'last') {
+            if (result.result.timedtexttracks.find(t=>t.rank==0).language === gRenderOptions.secondaryLanguageLastUsed) {
+              result.result.recommendedMedia.timedTextTrackId = result.result.timedtexttracks.find(t=>t.rank==1).new_track_id;
+              console.log("Restored recommended primary subtitle");
+            }
+          }
```

Comment:
Setting `result.result.recommendedMedia.timedTextTrackId` to the next recommended rank isn't always working as intended. There can be CC and non-CC captions in the new manifest, and then the original recommended sub will have rank 2. 

See screenshot: Polish was the original primary language, English is the `secondaryLanguageLastUsed`. English is in rank 0 and 1, so English is incorrectly activated again. We need another way to determine what the original primary subtitle language was.
![image](https://user-images.githubusercontent.com/13658335/207459153-db0b42da-7ab1-4762-86fd-671667d09882.png)

Finding the last occurrence of the language and then doing +1 is also not the solution, because I also want it to remember the Off state. The current approach always activates a language and doesn't remember Off.

I don't have time to investigate myself atm, so if you can't come up with another way now I would prefer to take this out and try adding it again in the future.

Reply:
The simplest way would be adding it to the settings dictionary so that it persists between sessions. But my preference would be a solution that only uses a global variable that also persists between sessions. Check for the original `recommendedMedia` when the show/movie is loaded, and then store that for the current session. But idk if this is possible, because once you request a new `preferredTextLocale` it seems to also overwrite your preference in the Netflix backend.

Reply:
I was wrong. There is no way to get original `recommendedMedia` if we set `preferredTextLocale` on manifest request. I've removed these changes.